R/retrieve.R

```diff
@@ -1025,8 +1025,11 @@ renv_retrieve_successful <- function(record, path, install = TRUE) {
     renv_retrieve_handle_remotes(record, subdir = subdir)
 
   # ensure its dependencies are retrieved as well
-  if (state$recursive)
+  if (state$recursive) local({
+    repos <- if (is.null(desc$biocViews)) getOption("repos") else renv_bioconductor_repos()
+    renv_scope_options(repos = repos)
     renv_retrieve_successful_recurse(deps)
+  })
```

Comment:
I'd personally write this as 

```suggestion
  if (state$recursive && !is.null(desc$biocViews)) { 
    local({
      renv_scope_options(repos = renv_bioconductor_repos())
      renv_retrieve_successful_recurse(deps)
    })
  }
```

I think `renv_bioconductor_repos()` probably needs a comment stating that it prepends to the existing repos; otherwise you have to follow quite a few calls to confirm it.

It also took me a bit of reading to reassure myself that if BiocManager isn't already installed, it will get installed in the right place. (It's also a bit unfortunate we'll install that in the middle of retrieving without any user notification, but I don't see how we can avoid that until we move to pak.)

Reply:
But `renv_retrieve_successful_recurse(deps)` needs to run regardless of the value of `biocViews`, hence the implementation I chose here.